tests/unit/test_config.py

```diff
@@ -73,29 +74,43 @@ def test_build_config_without_s3_data(self):
 
     def test_build_s3_storage_config(self):
         raw_s3_config_data = {
-            "endpoint": "s3.com:port",
+            "endpoint": "https://s3.com:port",
             "access-key": "your_access_key",
             "secret-key": "your_secret_key",
             "bucket": "your_bucket",
             "region": "your_region",
         }
         s3_config_data = _S3ConfigData(**raw_s3_config_data)
         s3_storage_config = self.coordinator._build_s3_storage_config(s3_config_data)
-        expected_config = {
+        expected_config_https = {
             "backend": "s3",
             "s3": {
                 "endpoint": "s3.com:port",
                 "access_key_id": "your_access_key",
                 "secret_access_key": "your_secret_key",
                 "bucket_name": "your_bucket",
                 "region": "your_region",
+                "insecure": "false",
             },
         }
-        self.assertEqual(s3_storage_config, expected_config)
+        self.assertEqual(DeepDiff(s3_storage_config, expected_config_https), {})
```

Comment:
Seems like this test is checking something like `d == Obj(d).d`.
Should the assert be reduced to:
```python
self.assertFalse(s3_storage_config["s3"]["insecure"])
```
?

Reply:
Not really, because I also want to test something that was not tested before which is the `endpoint` without the scheme.

`DeepDiff` returns an empty dict if there is no difference between the dicts.